firedrake/mesh.py

```diff
@@ -429,56 +358,82 @@ def callback(self):
-                                                         name="Coordinates")
-                else:
-                    self._coordinate_fs = functionspace.VectorFunctionSpace(self, "Lagrange", 1)
-
-                    coordinates = dmplex.reordered_coords(self._plex, self._coordinate_fs._global_numbering,
-                                                          (self.num_vertices(), geometric_dim))
-                    self.coordinates = function.Function(self._coordinate_fs,
-                                                         val=coordinates,
-                                                         name="Coordinates")
-            self._ufl_domain = ufl.Domain(self.coordinates)
-            # Build a new ufl element for this function space with the
-            # correct domain.  This is necessary since this function space
-            # is in the cache and will be picked up by later
-            # VectorFunctionSpace construction.
-            self._coordinate_fs._ufl_element = self._coordinate_fs.ufl_element().reconstruct(domain=self.ufl_domain())
-            # HACK alert!
-            # Replace coordinate Function by one that has a real domain on it (but don't copy values)
-            self.coordinates = function.Function(self._coordinate_fs, val=self.coordinates.dat)
-            # Add subdomain_data to the measure objects we store with
-            # the mesh.  These are weakrefs for consistency with the
-            # "global" measure objects
-            self._dx = ufl.Measure('cell', subdomain_data=weakref.ref(self.coordinates))
-            self._ds = ufl.Measure('exterior_facet', subdomain_data=weakref.ref(self.coordinates))
-            self._dS = ufl.Measure('interior_facet', subdomain_data=weakref.ref(self.coordinates))
-            # Set the subdomain_data on all the default measures to this
-            # coordinate field.
-            # We don't set the domain on the measure since this causes
-            # an uncollectable reference in the global space (dx is
-            # global).  Furthermore, it's never used anyway.
-            for measure in [ufl.dx, ufl.ds, ufl.dS]:
-                measure._subdomain_data = weakref.ref(self.coordinates)
         self._callback = callback
 
     def init(self):
-        """Finish the initialisation of the mesh.  Most of the time
-        this is carried out automatically, however, in some cases (for
-        example accessing a property of the mesh directly after
-        constructing it) you need to call this manually."""
+        """Finish the initialisation of the mesh."""
         if hasattr(self, '_callback'):
             self._callback(self)
 
+    @property
+    def topology(self):
+        """The underlying mesh topology object."""
+        return self
+
+    @property
+    def topological(self):
+        """Alias of topology.
+
+        This is to ensure consistent naming for some multigrid codes."""
+        return self
```

Comment:
I think the term *mesh topology* makes more sense than the term *topological mesh*, however, some multigrid codes require uniform naming for function, function space and mesh.